plugins/woocommerce/client/blocks/tests/e2e/utils/frontend/frontend-utils.page.ts

```diff
@@ -21,15 +21,34 @@ export class FrontendUtils {
 		return this.page.locator( selector );
 	}
 
-	async addToCart( itemName = '' ) {
-		const cartResponsePromise = this.page.waitForResponse( ( response ) => {
-			const url = response.url();
-			return (
-				url.includes( 'cart' ) ||
-				url.includes( 'add_to_cart' ) ||
-				url.includes( 'batch' )
+	/**
+	 * Wait for a single cart-related request to complete
+	 */
+	async waitForCartRequests( timeout = 5000 ): Promise< void > {
+		try {
+			await this.page.waitForResponse(
+				( response ) => {
+					const url = response.url();
+					return (
+						( url.includes( '/cart' ) ||
+							url.includes( '/add_to_cart' ) ||
+							url.includes( '/batch' ) ) &&
+						response.status() < 400
+					);
+				},
+				{ timeout }
```

Comment:
I'm not a big fan of waiting for an invisible thing (a fetch request) in an E2E, I guess this is a further refactor to the existing function `addToCart` and there isn't a good stable way to get this signal from UI

Reply:
I tried to see how adding to cart can be identified at the UI level, but there is too much variation. We do wait for invisible things with other utilities like `waitForCheckoutToFinishUpdating`